crates/rpc/rpc/src/eth/revm_utils.rs

```diff
@@ -227,6 +227,12 @@ where
 
     let mut env = build_call_evm_env(cfg, block, request)?;
 
+    if env.tx.gas_price == U256::ZERO {
```

Comment:
Do we want to go with the Erigon way here? This seems like a very weird thing to do.